tfhe/src/integer/ciphertext/glwe_ct_list.rs

```diff
@@ -0,0 +1,199 @@
+use super::{DataKind, Expandable, RadixCiphertext, SignedRadixCiphertext};
+use crate::integer::BooleanBlock;
+use crate::shortint::ciphertext::GlwePackedCiphertextList;
+use crate::shortint::glwe_compression::{GlweCompressionKey, GlweDecompressionKey};
+use crate::shortint::Ciphertext;
+use rayon::prelude::*;
+use serde::{Deserialize, Serialize};
+
+pub trait GlwePackable {
+    fn compact_into(self, messages: &mut Vec<Ciphertext>) -> DataKind;
+}
+
+impl GlwePackable for BooleanBlock {
+    fn compact_into(self, messages: &mut Vec<Ciphertext>) -> DataKind {
+        messages.push(self.0);
+        DataKind::Boolean
+    }
+}
+
+impl GlwePackable for RadixCiphertext {
+    fn compact_into(self, messages: &mut Vec<Ciphertext>) -> DataKind {
+        let num_blocks = self.blocks.len();
+
+        for block in self.blocks {
+            messages.push(block);
+        }
+
+        DataKind::Unsigned(num_blocks)
+    }
+}
+
+impl GlwePackable for SignedRadixCiphertext {
+    fn compact_into(self, messages: &mut Vec<Ciphertext>) -> DataKind {
+        let num_blocks = self.blocks.len();
+
+        for block in self.blocks {
+            messages.push(block);
+        }
+
+        DataKind::Signed(num_blocks)
+    }
+}
+
+pub struct GlwePackedCompressedCiphertextListBuilder {
+    pub(crate) ciphertexts: Vec<Ciphertext>,
+    pub(crate) info: Vec<DataKind>,
+}
+
+impl GlwePackedCompressedCiphertextListBuilder {
+    #[allow(clippy::new_without_default)]
+    pub fn new() -> Self {
+        Self {
+            ciphertexts: vec![],
+            info: vec![],
+        }
+    }
+
+    pub fn push<T>(&mut self, data: T) -> &mut Self
+    where
+        T: GlwePackable,
+    {
+        let n = self.ciphertexts.len();
+        let kind = data.compact_into(&mut self.ciphertexts);
+        assert_eq!(n + kind.num_blocks(), self.ciphertexts.len());
+
+        if kind.num_blocks() != 0 {
+            self.info.push(kind);
+        }
+
+        self
+    }
+
+    pub fn extend<T>(&mut self, values: impl Iterator<Item = T>) -> &mut Self
+    where
+        T: GlwePackable,
+    {
+        for value in values {
+            self.push(value);
+        }
+        self
+    }
+
+    pub fn build(&self, comp_key: &GlweCompressionKey) -> GlwePackedCompressedCiphertextList {
+        let packed_list = comp_key.pack_lwe_ciphertexts_into_glwes(&self.ciphertexts);
+
+        GlwePackedCompressedCiphertextList {
+            packed_list,
+            info: self.info.clone(),
+        }
+    }
+}
+
+#[derive(Clone, Serialize, Deserialize)]
+pub struct GlwePackedCompressedCiphertextList {
```

Comment:
GlwePacked -> rename to mathc CompressedCiphertextList

Reply:
Renamed `CompressedList`

Reply:
Ciphertext in the name ?

Reply:
Renamed `CompressedCiphertextList`